### ISSUE TITLE
Support ADC auth in SecretProviderClass

### ISSUE THREAD

**Issue by tam7t** [GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp#101]
### TL;DR
Allow the GCP plugin to obtain and use minikube auth.

By default the driver will trade the pod being scheduled's K8s SA for a GCP IAM token following the workload identity process. This does not work well if you're using minikube (GCP IAM doesn't know about your minikube workload pool).

minikube has a [gcp-auth](https://minikube.sigs.k8s.io/docs/handbook/addons/gcp-auth/) plugin that will put `GOOGLE_APPLICATION_CREDENTIALS` into pods. Adding a configuration to allow the GCP plugin to look for its own ADC creds would allow workloads scheduled to minikube to use the users's identity when accessing secrets.

### Design

**Proposal**

Introduce an `auth` parameter to the SecretProviderClass to tell the driver to pickup the Application Default Credentials of the plugin pod itself (not the pod being scheduled).

```
apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
kind: SecretProviderClass
metadata:
  name: app-secrets
spec:
  provider: gcp
  parameters:
    auth: self-adc
    secrets: |
      - resourceName: "projects/$PROJECT_ID/secrets/testsecret/versions/latest"
        fileName: "good1.txt"
```

Documentation around this feature would need to be careful to explain that it should not be used in a production cluster, as it would provide no separation between K8s namespaces.

**Alternatives considered**

An alternative is to the CSI Driver `nodePublishSecretRef` secret, where an exported SA json can be provided to the gcp provider for auth. This is used in CI where workload identity and workload federation are not options.

Do not want to encourage additional exports of SA creds though.

**Resources**

**Additional information**

**Comment by neuromantik33** [comment]
Sorry to chime in, but is this plugin only functional on GKE clusters with workload identity enabled? So on-prem non-anthos clusters are unable to use gcp as a provider?

**Comment by tam7t** [comment]
Hey @neuromantik33 I believe that if https://cloud.google.com/iam/docs/workload-identity-federation is configured that this would work without any additional exporting of SA creds.

The plugin currently obtains a K8s auth token for the scheduled pod, then uses that token to authenticate to IAM to obtain a GCP SA credentials. When Workload Identity Federation is configured for a cluster, then I _believe_ that the k8s SA token <-> GCP SA token transfer will work (though I have not tried this yet).

**Comment by tam7t** [comment]
This is now included in [v0.5.0](https://github.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp/blob/main/CHANGELOG.md#v050)
